Compass/ParseServer.swift

```diff
@@ -82,23 +149,122 @@ class PServer {
                     print("object doesnt exist")
                     return
                 }
-
-            newUserData["Location"] = PFGeoPoint(location: uData.location)
+            newUserData["Location"] = PFGeoPoint(location: location)
+            newUserData["UserName"] = PFUser.current()?.username
             newUserData.saveInBackground(block: { (sucess, error) in
                 guard error == nil else {
                     print(error ?? "Failed to update data")
                     return
                 }
-
             })
         })
     }
+    
+    func fetchUserList(completition: @escaping ([String]?, Error?) -> Void) {
+        let query = PFUser.query()
+        query?.findObjectsInBackground(block: { (objects, error) in
+            var result = [String]()
+            guard error == nil else {
+                completition(nil, error)
+                return
+            }
+            if let users = objects {
+                for object in users {
+                    if let user = object as? PFUser {
+                        if user.username != PFUser.current()?.username {
+                            result.append(user.username!)
+                        }
+                    }
+                }
+                completition(result, nil)
+            }
+        })
+    }
+    
+    func fetchUserData(userName: String, completion: @escaping (UserData?, Error?) -> Void) {
+            let result = UserData()
+            result.name = userName
+            let query = PFUser.query()
+            query?.findObjectsInBackground(block: { (objects, error) in
+                guard error == nil else {
+                    print(error ?? "error while fetching user names")
+                    completion(nil, error)
+                    return
+                }
+                guard let users = objects else {
+                    completion(nil, nil)
+                    return
+                }
+                    for object in users {
+                        guard let user = object as? PFUser else {
+                            completion(nil, nil)
+                            return
+                        }
+                        if user.username == userName {
```

Comment:
Presumably usernames are unique so this only happens once? Even so, I'd make the loop break once it's found the correct user

Reply:
Sure, I forgot to call return after the completion handler, also I need one more completion handler in case of a failed unwrap